### PR TITLE
Apache httpd w/TLS 1.3 support

### PR DESCRIPTION
**Pull request by dgarske**
* Add support for TLS v1.3 compatibility API `SSL_verify_client_post_handshake` for the server-side to support rehandshake (Required for Apache v2.4.39 with TLS v1.3)
* Added `SSL_CTX_set_post_handshake_auth` and `SSL_set_post_handshake_auth` API's for enabling or disabling post handshake authentication for TLS v1.3
* Add `OPENSSL_init_crypto` and `OPENSSL_init_ssl` API's.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -10357,6 +10357,30 @@ void wolfSSL_set_verify_result(WOLFSSL *ssl, long v)
...
+    int ret = NOT_COMPILED_IN;
+#if defined(WOLFSSL_TLS13) && defined(WOLFSSL_POST_HANDSHAKE_AUTH) && \
+    (!defined(NO_WOLFSSL_SERVER) || !defined(NO_WOLFSSL_CLIENT))
+    #ifndef NO_WOLFSSL_SERVER
+    if (ssl->options.side == WOLFSSL_SERVER_END) {
+        ret = wolfSSL_request_certificate(ssl);
+    }
+    #endif
+    #ifndef NO_WOLFSSL_CLIENT
+    if (ssl->options.side == WOLFSSL_CLIENT_END) {
```
Comment (SparkiDev) [inline src/ssl.c:10372]: OpenSSL returns an error when client side.

File: src/ssl.c [hunk 2]
```diff
@@ -10357,6 +10357,30 @@ void wolfSSL_set_verify_result(WOLFSSL *ssl, long v)
 #endif
 }
 
+/* For TLS v1.3 perform rehandshake. Returns 1=WOLFSSL_SUCCESS or 0=WOLFSSL_FAILURE */
```
Comment (SparkiDev) [inline src/ssl.c:10360]: Not a rehandshake - sending handshake messages after handshake complete.

File: src/ssl.c [hunk 3]
```diff
@@ -10357,6 +10357,30 @@ void wolfSSL_set_verify_result(WOLFSSL *ssl, long v)
...
+#if defined(WOLFSSL_TLS13) && defined(WOLFSSL_POST_HANDSHAKE_AUTH) && \
+    (!defined(NO_WOLFSSL_SERVER) || !defined(NO_WOLFSSL_CLIENT))
+    #ifndef NO_WOLFSSL_SERVER
+    if (ssl->options.side == WOLFSSL_SERVER_END) {
+        ret = wolfSSL_request_certificate(ssl);
+    }
+    #endif
+    #ifndef NO_WOLFSSL_CLIENT
+    if (ssl->options.side == WOLFSSL_CLIENT_END) {
+        ret = wolfSSL_allow_post_handshake_auth(ssl);
```
Comment (SparkiDev) [inline src/ssl.c:10373]: Looks like SSL_[CTX_]set_post_handshake_auth() enables/disables.

**Review by SparkiDev — CHANGES_REQUESTED** [review]
Rest is fine.

File: src/ssl.c [hunk 4]
```diff
@@ -10357,6 +10357,34 @@ void wolfSSL_set_verify_result(WOLFSSL *ssl, long v)
...
+    defined(WOLFSSL_TLS13) && defined(WOLFSSL_POST_HANDSHAKE_AUTH)
+/* For TLS v1.3 send handshake messages after handshake completes. */
+/* Returns 1=WOLFSSL_SUCCESS or 0=WOLFSSL_FAILURE */
+int wolfSSL_verify_client_post_handshake(WOLFSSL* ssl)
+{
+    int ret = wolfSSL_request_certificate(ssl);
+    return (ret == 0) ? WOLFSSL_SUCCESS : WOLFSSL_FAILURE;
+}
+
+void wolfSSL_CTX_set_post_handshake_auth(WOLFSSL_CTX* ctx, int val)
```
Comment (haydenroche5) [inline src/ssl.c:10370]: I think the return type should be int for this guy and wolfSSL_set_post_handshake_auth.

**Pull request merged**